test/api_integration/services/supertest.ts

```diff
@@ -20,5 +21,16 @@ export function KibanaSupertestProvider({ getService }: FtrProviderContext) {
 export function ElasticsearchSupertestProvider({ getService }: FtrProviderContext) {
   const config = getService('config');
   const elasticSearchServerUrl = formatUrl(config.get('servers.elasticsearch'));
-  return supertestAsPromised(elasticSearchServerUrl);
+
+  let agentOptions = {};
+  const kbnServerArgs: string[] = config.get('kbnTestServer.serverArgs');
+  const esCa = kbnServerArgs
+    .find((arg) => arg.startsWith('--elasticsearch.ssl.certificateAuthorities'))
+    ?.split('=')[1];
+  if (esCa) {
+    agentOptions = { ca: readFileSync(esCa) };
+  }
```

Comment:
To be honest, I don't know if it is the best way, but I don't see any issue with it. Maybe it makes checking  `esCa` to be a valid filepath before reading?

Reply:
Thanks for the feedback @dmlemeshko!
I thought about guarding the file read, but because of the following two items decided against it:
- We read this value from the config, so whoever provided it is responsible for it. Also, I think Kibana wouldn't even start with a wrong value in that configuration.
- But given the first point is not valid, what would we do if it's not a valid file path or in case of missing permissions to read the file, etc.? We would probably throw an exception, so I kind of left it to `readFileSync` to throw in case things go wrong.

Having said that, I'm open to change that and add something like
```
try {
  read file
} catch {
  do something
}
```
but still need to figure out what `do something` would be. Ignore the error and continue without the CA is not ideal as failures would probably come up later in the process where it's harder to find out what went wrong.

Reply:
For Kibana we have `servers.kibana.certificateAuthorities` defined via the config file like so: https://github.com/elastic/kibana/blob/a0ce7b5aa887d34a7a892553c66f25d72e38d827/x-pack/test/functional_embedded/config.ts#L27 I recommend we do the same thing for `servers.elasticsearch` rather than trying to detect the CA from the CLI args.

Reply:
Thanks for the feedback @spalger!
I like that idea and applied it in 542a26e.